packages/backend-modules/voting/lib/Submission.js

```diff
@@ -117,38 +136,60 @@ const createSubmissionsQuery = ({
       ],
     },
   }
+
   const mustValue = value && {
-    term: { 'resolved.answers.payload.value': value },
-  }
-  const mustSubmissionId = filters?.id && { term: { id: filters.id } }
-  const mustNotSubmissionId = filters?.not && { term: { id: filters.not } }
-  const mustSubmissionIds = filters?.submissionIds?.length && {
-    terms: { id: filters.submissionIds },
-  }
-  const mustNotSubmissionIds = filters?.notSubmissionIds?.length && {
-    terms: { id: filters.notSubmissionIds },
+    nested: {
+      path: 'resolved.answers',
+      query: { term: { 'resolved.answers.payload.value': value } },
+    },
   }
-  const mustAnsweredQuestionIds = filters?.answeredQuestionIds && {
+
+  const expectedAnswers = (filters?.answeredQuestionIds || [])
```

Comment:
Is this how you filter for only the answers to the questions you want, and not all answers in the answer sets which contain the question to be answered?

Reply:
It uses array of strings – question Ids – and foreces them to an array of objects `{ questionId: '123123' }` – using shape of `SubmissionFilterAnswer`.

It then concatinates this array of objects with objects in `filters.answers`, again, in the shape of `SubmissionFilterAnswer`.

Example:

```graphql
filters: {
  answeredQuestionIds: ['123']
  answers: [
    { questionId: '456' },
    { questionId: '789', valueLength: { max: 100 } }
  ]
}
```

will become:

```graphql
[
    { questionId: '123' },
    { questionId: '456' },
    { questionId: '789', valueLength: { max: 100 } }
]
```